go/arrow/ipc/file_reader.go

```diff
@@ -337,6 +341,11 @@ func newRecord(schema *arrow.Schema, meta *memory.Buffer, body ReadAtSeeker) arr
 	for i, field := range schema.Fields() {
 		cols[i] = ctx.loadArray(field.Type)
 	}
+	defer func() {
+		for _, c := range cols {
+			c.Release()
+		}
+	}()
```

Comment:
```go
for i, field := range schema.Fields() {
        cols[i] = ctx.loadArray(field.Type)
        defer cols[i].Release()
}
```

No need for the separate defer func and loop

Reply:
I didn't want to call `defer` inside a loop - is that ok?

Reply:
yup, it's perfectly okay. Remember that defer's only run at the end of the enclosing *function* not scope.